common/contract.ts

```diff
@@ -107,6 +111,21 @@ export type CPMM = {
   }
 }
 
+export type Uniswap2 = {
+  mechanism: 'uniswap-2'
+  // outcome can be e.g. 'M$' or a 'SHARE'
+  pool: { [outcome: string]: number }
```

Comment:
Why not just use fields for shares and mana? Multivariable case probably is not very common.

Reply:
Wanted to leave open the mechanism in the future to eg swap two different shares/coins, I think? But maybe YAGNI. I don't think it matters much either way, here.